src/liboslexec/shadingsys.cpp

```diff
@@ -4174,15 +4174,17 @@ osl_range_check_err (int indexvalue, int length, const char *symname,
 
 
 
+// Return the raytype bit corresponding to a raytype name.
 // Asked if the raytype is a name we can't know until mid-shader.
-OSL_SHADEOP int osl_raytype_name (void *sg_, void *name)
+// FIXME: We should pass the context, not the sg.
```

Comment:
Why a FIXME, any technical barrier to passing context?  For that matter, I've been pondering a ReadOnlyContext or DeviceContext that might be a stripped down version of the full context to answer just these types of questions.